File: packages/java/src/main/java/com/ummalqura/MonthInfo.java
package com.ummalqura;

/**
 * Month information from the data table
 */
public class MonthInfo {
    private final int hijriYear;
    private final int hijriMonth;
    private final int monthLength;
    private final int firstDayJdn;

    public MonthInfo(int hijriYear, int hijriMonth, int monthLength, int firstDayJdn) {
        this.hijriYear = hijriYear;
        this.hijriMonth = hijriMonth;
        this.monthLength = monthLength;
        this.firstDayJdn = firstDayJdn;
    }

    public int getHijriYear() { return hijriYear; }
    public int getHijriMonth() { return hijriMonth; }
    public int getMonthLength() { return monthLength; }
    public int getFirstDayJdn() { return firstDayJdn; }
}

File: packages/php/src/MonthInfo.php
<?php

namespace UmmAlQura;

class MonthInfo
{
    public function __construct(
        private int $hijriYear,
        private int $hijriMonth,
        private int $monthLength,
        private int $firstDayJdn
    ) {}

    public function getHijriYear(): int { return $this->hijriYear; }
    public function getHijriMonth(): int { return $this->hijriMonth; }
    public function getMonthLength(): int { return $this->monthLength; }
    public function getFirstDayJdn(): int { return $this->firstDayJdn; }

    public function toArray(): array
    {
        return [
            'hijri_year' => $this->hijriYear,
            'hijri_month' => $this->hijriMonth,
            'month_length' => $this->monthLength,
            'first_day_jdn' => $this->firstDayJdn,
        ];
    }
}
